File: yht0827/1월/1월 9일/이진 변환 반복하기.java
import java.util.*;

class Solution {
	public int countChar(String str, char ch) {
		return str.length() - str.replace(String.valueOf(ch), "").length();
	}

	public int[] solution(String s) {
		int[] answer = new int[2];

		int count = 0;
		int zero = 0;

		while (!s.equals("1")) {

			count += 1;
			zero += countChar(s, '0');

			s = s.replace("0", "");

			s = Integer.toBinaryString(s.length());

		}

		answer[0] = count;
		answer[1] = zero;

		return answer;
	}
}

File: Eunleelee/2024-01/1월 9일/이진_변환_반복하기.py
def solution(x):
    zero_num = 0
    count = 0
    
    while x != '1':
        zero_num += x.count('0')
        x = x.replace('0', '')
        x = format(len(x), 'b')
        count += 1
    
    answer = [count, zero_num]
    return answer

print(solution("01110"))